### PR TITLE
Give wrestlers a randomised outfit

### PR DESCRIPTION
**Pull request by flappybatpal**
## About the PR 
Adds a way to pick random types from a base type, thanks Zewaka.
Assigns wrestlers a random outfit, with a 50% chance of luchador or American/Japanese styling plus a cosmetic championship belt.
Note this does remove their ID card which will make their life harder.

## Why's this needed? 
At the moment wrestlers lack any sort of identity, essentially being a staff assistant with innate wrestle belt abilities. 
This does make them less subtle but we already try to give them a name ending in "the Wrestler" plus the constant belt style flex emoting so I would hardly call them subtle as-is.

## Changelog 

```changelog
(u)Flappybat
(*)Wrestler antagonists now start with an appropriate outfit.
```

### CODE REVIEW

File: code/obj/item/clothing/masks.dm [hunk 1]
```diff
@@ -559,22 +559,27 @@ TYPEINFO(/obj/item/clothing/mask/monkey_translator)
...
 
-	green
+/obj/item/clothing/mask/wrestling/green
 		icon_state = "greenmask"
 		item_state = "greenmask"
 
-	blue
+/obj/item/clothing/mask/wrestling/blue
 		icon_state = "bluemask"
 		item_state = "bluemask"
```
Comment (ZeWaka) [inline code/obj/item/clothing/masks.dm:584]: broken indentation

File: code/obj/item/clothing/uniforms.dm [hunk 2]
```diff
@@ -814,55 +822,59 @@
...
+	icon_state = "shortsP"
+	item_state = "shortsP"
+
+/obj/item/clothing/under/shorts/black
+	icon_state = "shortsB"
+	item_state = "shortsB"
+
+/obj/item/clothing/under/shorts/psyche
+	name = "psychedelic shorts"
+	desc = "Only wear these if you don't mind people staring at your crotch."
```
[the remark this answers is not in the document]
Comment (ZeWaka) [reply code/obj/item/clothing/uniforms.dm:847]: yeah....

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/wrestler/wrestler.dm [hunk 3]
```diff
@@ -42,6 +42,40 @@
 			src.ability_holder.addAbility(/datum/targetable/wrestler/throw)
 			src.ability_holder.addAbility(/datum/targetable/wrestler/slam)
 
+		// Assign wrestle attire.
+		if (ishuman(src.owner.current))
```
[the remark this answers is not in the document]
Comment (flappybatpal) [reply code/modules/antagonists/wrestler/wrestler.dm:17]: Added below the wrestle doodle where I think you meant.

File: code/modules/antagonists/wrestler/wrestler.dm [hunk 4]
```diff
@@ -42,6 +42,40 @@
...
+				H.equip_new_if_possible(rand_shorts, H.slot_w_uniform)
+			else
+				if (prob(50))
+					var/obj/item/clothing/under/shorts/rand_shorts = get_random_atom(/obj/item/clothing/under/shorts)
+					H.equip_new_if_possible(rand_shorts, H.slot_w_uniform)
+				else
+					H.equip_new_if_possible(/obj/item/clothing/under/gimmick/macho/random_color, H.slot_w_uniform)
+				if (prob(33))
+					H.equip_new_if_possible(/obj/item/clothing/head/bandana/random_color, H.slot_head)
+			var/shoes_prob = rand(1, 100)
```
[the remark this answers is not in the document]
Comment (flappybatpal) [reply code/modules/antagonists/wrestler/wrestler.dm:36]: Thanks this is a cleaner approach I hadn't considered.

File: _std/types.dm [hunk 5]
```diff
@@ -466,3 +467,11 @@ proc/istypes(datum/dat, list/types)
 		if(istype(dat, type))
 			return TRUE
 	return FALSE
+
+/proc/get_random_atom(atom_type, return_instance = FALSE, return_instance_newargs = null)
```
[the remark this answers is not in the document]
Comment (flappybatpal) [reply _std/types.dm:471]: I don't know how spaces got there, fixed

File: code/obj/item/clothing/masks.dm [hunk 6]
```diff
@@ -559,22 +559,27 @@ TYPEINFO(/obj/item/clothing/mask/monkey_translator)
 	item_state = "melons"
 	see_face = 0
 
+TYPEINFO(/obj/item/clothing/mask/wrestling)
+   random_subtypes = list(/obj/item/clothing/mask/wrestling,
```
[the remark this answers is not in the document]
Comment (flappybatpal) [reply code/obj/item/clothing/masks.dm]: Huh, yeah also noted the style for other multi line lists is just one indent. Fixed.

File: code/obj/item/clothing/uniforms.dm [hunk 7]
```diff
@@ -805,6 +805,14 @@
 
 // Athletic Gear
 
+TYPEINFO(/obj/item/clothing/under/shorts)
+   random_subtypes = list(/obj/item/clothing/under/shorts,
```
[the remark this answers is not in the document]
Comment (flappybatpal) [reply code/obj/item/clothing/uniforms.dm]: Fixed.

File: code/obj/item/clothing/uniforms.dm [hunk 8]
```diff
@@ -814,55 +822,59 @@
...
+	item_state = "shortsB"
+
+/obj/item/clothing/under/shorts/psyche
+	name = "psychedelic shorts"
+	desc = "Only wear these if you don't mind people staring at your crotch."
+	icon_state = "shortsPs"
+	item_state = "shortsPs"
+
+TYPEINFO(/obj/item/clothing/under/shorts/luchador)
+   random_subtypes = list(/obj/item/clothing/under/shorts/luchador,
```
[the remark this answers is not in the document]
Comment (flappybatpal) [reply code/obj/item/clothing/uniforms.dm]: Fixed.

File: code/obj/item/clothing/uniforms.dm [hunk 9]
```diff
@@ -805,6 +805,14 @@
 
 // Athletic Gear
 
+TYPEINFO(/obj/item/clothing/under/shorts)
+  random_subtypes = list(/obj/item/clothing/under/shorts,
```
Comment (FlameArrow57) [inline code/obj/item/clothing/uniforms.dm:809]:
```suggestion
	random_subtypes = list(/obj/item/clothing/under/shorts,
```

File: code/modules/antagonists/wrestler/wrestler.dm [hunk 10]
```diff
@@ -9,9 +9,43 @@
...
 	give_equipment(fake_equipment = FALSE)
 		src.fake = fake_equipment
+		src.owner.current.add_wrestle_powers(fake_equipment)
 		if (ismobcritter(src))
 			display_name = "wrestledoodle"
-		src.owner.current.add_wrestle_powers(fake_equipment)
+			return
+
+		// Assign wrestle attire.
+		if (ishuman(src.owner.current))
```
Comment (FlameArrow57) [inline code/modules/antagonists/wrestler/wrestler.dm:18]: Oh that's fine too I think, was initially meaning you could also do if (!ishuman(src.owner.current)) return

File: code/modules/antagonists/wrestler/wrestler.dm [hunk 11]
```diff
@@ -9,9 +9,43 @@
...
+			else
+				if (prob(50))
+					var/obj/item/clothing/under/shorts/rand_shorts = get_random_atom(/obj/item/clothing/under/shorts)
+					H.equip_new_if_possible(rand_shorts, H.slot_w_uniform)
+				else
+					H.equip_new_if_possible(/obj/item/clothing/under/gimmick/macho/random_color, H.slot_w_uniform)
+				if (prob(33))
+					H.equip_new_if_possible(/obj/item/clothing/head/bandana/random_color, H.slot_head)
+			var/obj/item/clothing/under/shorts/rand_shoes
+			switch(pick(1,2,3,4))
```
Comment (FlameArrow57) [inline code/modules/antagonists/wrestler/wrestler.dm:38]:
```suggestion
			switch(pick(1, 2, 3, 4))
```

File: code/modules/antagonists/macho_man/abilities/_macho_man_mob.dm [hunk 12]
```diff
@@ -421,6 +421,14 @@ var/list/snd_macho_idle = list('sound/voice/macho/macho_alert16.ogg', 'sound/voi
 	icon_state = "machopants"
 	item_state = "machopants"
 
+	random_color
+	icon_state = "machopants_base"
+	item_state = "machopants_base"
+
+	New()
+		..()
+		src.color = random_saturated_hex_color(1)
```
Comment (FlameArrow57) [inline code/modules/antagonists/macho_man/abilities/_macho_man_mob.dm:430]:
```suggestion
		icon_state = "machopants_base"
		item_state = "machopants_base"

		New()
			..()
			src.color = random_saturated_hex_color(1)
```
Thinking indent was meant here?

File: code/obj/item/clothing/uniforms.dm (hunk 9 above)
Comment (flappybatpal) [reply code/obj/item/clothing/uniforms.dm:809]: I'm really confused as to how spaces keep getting in here, it's not an issue I've had before.

**Comment by FlameArrow57** [comment]
Nothing else in mind

File: _std/types.dm [hunk 13]
```diff
@@ -233,6 +233,7 @@ var/list/list/by_cat = list()
 
 /typeinfo/atom
 	parent_type = /typeinfo/datum
+	var/random_subtypes = null
```
Comment (ZeWaka) [inline _std/types.dm:236]: TODO: put in code guide

**Review by ZeWaka — APPROVED** [review]

File: _std/types.dm [hunk 14]
```diff
@@ -466,3 +467,10 @@ proc/istypes(datum/dat, list/types)
 		if(istype(dat, type))
 			return TRUE
 	return FALSE
+
```
Comment (flrsh) [inline _std/types.dm:470]: Add code doc.

File: _std/types.dm [hunk 15]
```diff
@@ -466,3 +467,10 @@ proc/istypes(datum/dat, list/types)
...
+/proc/get_random_atom(atom_type, return_instance = FALSE, return_instance_newargs = null)
```
Comment (flrsh) [inline _std/types.dm:471]: `get_random_subtype` seems more fitting?

File: _std/types.dm (hunk 13 above)
Comment (flrsh) [inline _std/types.dm:236]: Add code doc; it's unclear what `random_subtypes` actually means and how it should be used.

**Review by flrsh — CHANGES_REQUESTED** [review]

**Comment by ZeWaka** [comment]
.

**Review by flrsh — APPROVED** [review]

**Pull request merged**